src/custom-blocks/auto-item-list/index.js

```diff
@@ -0,0 +1,225 @@
+import { store as coreStore } from '@wordpress/core-data';
+
+const { __ } = wp.i18n;
+const { registerBlockType, registerBlockStyle } = wp.blocks;
+const { Fragment } = wp.element;
+const { useSelect } = wp.data;
+const { InspectorControls, MediaUpload, InnerBlocks } = wp.blockEditor;
+const { PanelBody, SelectControl, Button, TextControl, ToggleControl, RadioControl} = wp.components;
+const allowedMediaTypes = ['image'];
+
+import { __experimentalNumberControl as NumberControl } from '@wordpress/components';
+import { __experimentalText as Text } from '@wordpress/components';
+
+registerBlockType("mojblocks/auto-item-list", {
+  title: __("Items List", "mojblocks"),
+  description: __('Auto-populates with most recent or next items based on a date'),
+  category: "mojblocks",
+  icon: "slides",
+  keywords: [__('latest item'), __('latest news'), __('recent items'), __('recent news'),  __('next items'), __('headlines')],
+
+  attributes: {
+    listHasDate: {
+      type: "boolean",
+      default: true
+    },
+    listEmptyText: {
+      type: "string",
+      default: "No items to display."
+    },
+    listItemType: {
+      type: "string",
+      default: "post"
+    },
+    listClassName: {
+      type: 'string'
+    }
+
+  },
+
+  edit: props => {
+    const {
+      setAttributes,
+      attributes: {
+        listEmptyText,
+        listHasDate,
+        listItemType
+      },
+      className
+    } = props
+
+    const {
+      allPostTypes,
+    } = useSelect(
+      ( select ) => {
+          const { getPostTypes } = select(
+          coreStore
+        );
+        const allPostTypesList = getPostTypes(
+          { per_page: -1 }
+        );
+        return {
+          allPostTypes: allPostTypesList
+        };
+      }
+    );
+
+    let itemTypes = [{
+      label: "-",
+      value: ""
+    }]
+    let restrctedPostTypes = ["Media", "Posts", "Pages"];
+    let itemTypesFinishedParsing = false;
+
+    if (allPostTypes) {
+      allPostTypes.forEach(thisPostType => {
+
+        //Prevents core post types to be shown but still allows CPTs which do not have single views
+        if (!restrctedPostTypes.includes(thisPostType.name) && thisPostType.visibility.show_ui && thisPostType.visibility.show_in_nav_menus) {
+          itemTypes.push({
+            label: thisPostType.name,
+            value: thisPostType.slug
+          });
+        }
+      });
+      itemTypesFinishedParsing = true;
+    }
+
+    const {
+      allDocuments,
+    } = useSelect(
+      ( select ) => {
+        if (listItemType) {
+
+          const { getEntityRecords } = select(
+            coreStore
+          );
+
+          const posts = getEntityRecords(
+            'postType',
+            listItemType,
+            { per_page: -1 }
+          );
+
+          return {
+            allDocuments: posts
+          };
+        } else {
+          return {
+            allDocuments: false
+          };
+        }
+      }
+    );
```

Comment:
Similar to the all posts above, add a fallback for empty posts or none and make more efficent so only runs when things change ie, use React hook to check changes.
```
const { allDocuments } = useSelect(
    (select) => {
        if (listItemType) {
            const { getEntityRecords } = select(coreStore);
            const posts = getEntityRecords('postType', listItemType, { per_page: -1 });

            return {
                allDocuments: posts || [] // fallback
            };
        }

        return {
            allDocuments: []
        };
    },
    [listItemType] // Re-run the hook if listItemType changes
);
```